packages/app-accounts/src/Editor.tsx

```diff
@@ -85,150 +75,54 @@ class Editor extends React.PureComponent<Props, State> {
   }
 
   renderData () {
-    const { accountAll, t } = this.props;
-    const { current, editedName } = this.state;
-
-    if (!accountAll || !Object.keys(accountAll).length) {
-      return t('editor.none', {
-        defaultValue: 'There are no saved accounts. Add some first.'
-      });
-    }
+    const { current, editedName, onChangeAccount, onChangeName, t } = this.props;
 
-    const address = current
-      ? current.address()
-      : undefined;
+    const address = current ? current.address() : undefined;
 
     return (
-      <div className='ui--grid'>
-        <AddressSummary
-          className='shrink'
-          value={address || ''}
-        />
-        <div className='grow'>
-          <div className='ui--row'>
+      <div className='accounts--flex-group-row'>
+        <div className='accounts--flex-container-col-summary'>
+          <div className='accounts--flex-item'>
+            <AddressSummary
+              buttonChildren={<DownloadButton address={address} />}
+              className='shrink'
+              value={address}
+            />
+          </div>
+        </div>
+        <div className='accounts--flex-container-col-inputs'>
+          <div className='accounts--flex-item'>
             <InputAddress
               className='full'
               hideAddress
               isInput={false}
               label={t('editor.select', {
                 defaultValue: 'using my account'
               })}
-              onChange={this.onChangeAccount}
+              onChange={onChangeAccount}
```

Comment:
I am not convinced about moving any of this logic to index.tsx - the base component does not care about editing, it is only concerned with rendering the correct Components.